protocol/wasmbinding/msg_plugin.go

```diff
@@ -140,3 +197,172 @@ func parseAddress(addr string) (sdk.AccAddress, error) {
+		order.GoodTilOneof = &clobtypes.Order_GoodTilBlock{
+			GoodTilBlock: placeOrder.Order.GoodTilBlock,
+		}
+		// throw error for short term contracts since they are not supported
+		return nil, nil, errors.New("short term orders are not supported for contracts")
+	} else {
+		order.GoodTilOneof = &clobtypes.Order_GoodTilBlockTime{
+			GoodTilBlockTime: lib.MustConvertIntegerToUint32(placeOrder.Order.GoodTilBlockTime),
+		}
+
+		if ctx.IsCheckTx() || ctx.IsReCheckTx() {
+			// We don't process stateful orders in CheckTx, so do nothing.
+			return nil, nil, nil
+		}
+
+		fmt.Printf("Placing stateful order: %+v\n", order)
+		processProposerMatchesEvents := m.clob.GetProcessProposerMatchesEvents(ctx)
+
+		// setting is internal order to false since this is being placed from a contract
+		if err := m.clob.PlaceStatefulOrder(ctx, &clobtypes.MsgPlaceOrder{Order: order}, false); err != nil {
+			return nil, nil, err
+		}
+
+		if order.IsConditionalOrder() {
+			m.clob.GetIndexerEventManager().AddTxnEvent(
+				ctx,
+				indexerevents.SubtypeStatefulOrder,
+				indexerevents.StatefulOrderEventVersion,
+				indexer_manager.GetBytes(
+					indexerevents.NewConditionalOrderPlacementEvent(
+						order,
+					),
+				),
+			)
+			processProposerMatchesEvents.PlacedConditionalOrderIds = append(
+				processProposerMatchesEvents.PlacedConditionalOrderIds,
+				order.OrderId,
+			)
+		} else {
+			m.clob.GetIndexerEventManager().AddTxnEvent(
+				ctx,
+				indexerevents.SubtypeStatefulOrder,
+				indexerevents.StatefulOrderEventVersion,
+				indexer_manager.GetBytes(
+					indexerevents.NewLongTermOrderPlacementEvent(
+						order,
+					),
+				),
+			)
+			processProposerMatchesEvents.PlacedLongTermOrderIds = append(
+				processProposerMatchesEvents.PlacedLongTermOrderIds,
+				order.OrderId,
+			)
+		}
+		m.clob.MustSetProcessProposerMatchesEvents(
+			ctx,
+			processProposerMatchesEvents,
+		)
+	}
+
+	return nil, nil, err
+}
+
+func (m *CustomMessenger) cancelOrder(
+	ctx sdk.Context,
+	contractAddr sdk.AccAddress,
+	cancelOrder *bindings.CancelOrder,
+) ([]sdk.Event, [][]byte, error) {
+	if cancelOrder == nil {
+		return nil, nil, wasmvmtypes.InvalidRequest{Err: "create transfer null transfer"}
+	}
+
+	address, err := parseAddress(cancelOrder.OrderId.SubaccountId.Owner)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	number := cancelOrder.OrderId.SubaccountId.Number
+
+	orderId := clobtypes.OrderId{
+		SubaccountId: satypes.SubaccountId{
+			Owner:  address.String(),
+			Number: number,
+		},
+		ClientId:   cancelOrder.OrderId.ClientId,
+		OrderFlags: cancelOrder.OrderId.OrderFlags,
+		ClobPairId: cancelOrder.OrderId.ClobPairId,
+	}
+
+	if cancelOrder.OrderId.OrderFlags == clobtypes.OrderIdFlags_ShortTerm {
+		// Send error if trying to cancel a short term order. Since short term orders
+		// are not supported for contracts
+		return nil, nil, errors.New("short term orders  are not supported for contracts")
+
+	} else {
```

Comment:
Why do we need the logic (from line 333 to 367) besides just calling the entrypoint `CancelStatefulOrder`?